### ISSUE TITLE
Catch make-instance non-class symbols at compile time

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#2262]
A frequent class of error is when calling `make-instance` over a symbol that's not a class.  This happens due to typos, renaming or, typically, when moving class definition around.

A trick could be to shadow `make-instance` with a macro that checks for class existence at macro-expansion time.
Proof of concept:

```lisp
(uiop:define-package nyxt
   (:use :cl) 
  (:shadow #:make-instance))

(in-package :nyxt)

(defmacro make-instance (sym &rest args)
   (find-class (second sym) )
 `(apply #'cl:make-instance ,sym ,args ))

(lambda () (make-instance 'i-do-not-exist))
; in: LAMBDA ()
;     (FOO::MAKE-INSTANCE 'FOO::I-DO-NOT-EXIST)
; 
; caught ERROR:
;   during macroexpansion of (MAKE-INSTANCE 'I-DO-NOT-EXIST). Use
;   *BREAK-ON-SIGNALS* to intercept.
;   
;    There is no class named FOO::I-DO-NOT-EXIST.
; 
; compilation unit finished
;   caught 1 ERROR condition
; => #<FUNCTION (LAMBDA ()) {547D097B}>
```

But really we only need this in the CI.  So I suggest we add a compilation option that defines the package and the macro only if some environment variable is set, and set this variable from the GitHub CI script.

Thoughts?

**Comment by aadcg** [comment]
Could you please provide a concrete example of the problem it tries to solve? Thanks.

**Comment by Ambrevar** [comment]
The following compiles without warning:

```lisp
(make-instance 'non-class-symbol)
```

I would like to report such mistakes.

**Comment by Ambrevar** [comment]
The fix in #2295 does not catch everything, namely it does not catch the slot initform errors.  Well, maybe it's a limitation of compilers and there is nothing we can do.
